ugvc/flow_format/flow_based_read.py

```diff
@@ -412,10 +419,22 @@ def _matrix_from_qual_tp(
             right_prob[right_neighbor >= len(key)] = 0
             flow_matrix[1, key == 0] = np.min(np.vstack((left_prob, right_prob)), axis=0)[key == 0]
 
-        place_to_locate = tp_tag + np.repeat(key, key.astype(int))
+        # repeat each value k (hmer-size) k times, e.g [0, 1, 2, 1, 3] -> [1, 2, 2, 1, 3, 3, 3]
+        # This results in hmer_sizes in base-space
+        # Notice a trimmed hmer will result in the wrong original hmer-size 
+        hmer_sizes = np.repeat(key, key.astype(int))
+        
+        # place_to_locate[i] is the hmer-size which the probability in base[i] refers to
+        # This size will be used to locate the probability in the flow-matrix
+        place_to_locate = tp_tag + hmer_sizes
+
         place_to_locate = np.clip(place_to_locate, None, max_hmer_size)
+        if spread_edge_probs:
+            place_to_locate[0] = np.clip(place_to_locate[0], 0, None)
```

Comment:
On a second thought, this prbably needs to apply to all the first/lst hmer bases. not just first/last base